packages/core/src/oidc-utils.ts

```diff
@@ -0,0 +1,120 @@
+import * as actions_http_client from '@actions/http-client'
+import {IHeaders,IRequestOptions} from '@actions/http-client/interfaces'
+import {HttpClient} from '@actions/http-client'
+import {BearerCredentialHandler} from '@actions/http-client/auth'
+import {debug, setSecret} from './core'
+
+interface IOidcClient {
+
+  createHttpClient(): actions_http_client.HttpClient
+
+  getApiVersion(): string
+
+  getRuntimeToken(): string
+
+  getIDTokenUrl(): string
+
+  isSuccessStatusCode(statusCode?: number): boolean
+
+  postCall(id_token_url: string, audience: string): Promise<string>
+
+  parseJson(body: string): string
+
+  getIDToken(audience: string): Promise<string>
+}
+
+export class OidcClient implements IOidcClient {
+
+  createHttpClient(allowRetry = true, maxRetry = 10) {
+    let requestOptions : IRequestOptions = {}
+    requestOptions.allowRetries = allowRetry
+    requestOptions.maxRetries = maxRetry
+    return new HttpClient('actions/oidc-client', [
+      new BearerCredentialHandler(this.getRuntimeToken())],
+      requestOptions)
+  }
+
+  getApiVersion(): string {
+    return '2.0'
+  }
+
+  getRuntimeToken(){
+    const token = process.env['ACTIONS_RUNTIME_TOKEN']
+    if (!token) {
+      throw new Error('Unable to get ACTIONS_RUNTIME_TOKEN env variable')
+    }
+    return token
+  }
+
+  getIDTokenUrl(){
+    let runtimeUrl = process.env['ACTIONS_ID_TOKEN_REQUEST_URL']
+    if (!runtimeUrl) {
+      throw new Error('Unable to get ACTIONS_ID_TOKEN_REQUEST_URL env variable')
+    }
+    return runtimeUrl + '?api-version=' + this.getApiVersion()
+  }
+
+  isSuccessStatusCode(statusCode?: number): boolean {
+    if (!statusCode) {
+      return false
+    }
+    return statusCode >= 200 && statusCode < 300
+  }
+
+  async postCall(id_token_url: string, audience: string): Promise<string> {
+
+    const httpclient = this.createHttpClient()
+    if (httpclient === undefined) {
+      throw new Error(`Failed to get Httpclient `)
+    }
+
+    let additionalHeaders: IHeaders = {}
+    additionalHeaders[actions_http_client.Headers.ContentType] = actions_http_client.MediaTypes.ApplicationJson
+    additionalHeaders[actions_http_client.Headers.Accept] = actions_http_client.MediaTypes.ApplicationJson
+
+    debug(`audience is ${audience !== null ? audience : 'null'}`)
+
+    const data: string = audience !== null ? JSON.stringify({aud: audience}) : ''
+    const response = await httpclient.post(id_token_url, data, additionalHeaders)
```

Comment:
I believe you can use the `postJson` method from HTTPClient and then you don't have to add these headers or stringify the payload.

Reply:
In fact, if we use the `postJson` method this entire `postCall` method might not be needed